### ISSUE TITLE
Missing values for columns "%gnice" or "%idle" sometimes.

### ISSUE THREAD

**Issue by ams-tschoening** [sysstat/sysstat#288]
Some of my SAR files of Ubuntu 18.04 contain lines like the following. Though, I think I have seen something similar with UB 16.04 as well and simply ignored it until now.

```
00:00:01        CPU      %usr     %nice      %sys   %iowait    %steal      %irq     %soft    %guest    %gnice     %idle
00:00:01        all      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00
00:00:01          0      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
00:00:01          1      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
00:00:01          2      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
00:00:01          3      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
00:00:01          4      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
00:00:01          5      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
00:00:01          6      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
00:00:01          7      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
00:05:01        all      4.49      0.00      1.01      6.60      0.00      0.00      0.09      0.00      0.00     87.80
00:05:01          0      1.13      0.00      0.82      1.47      0.00      0.00      0.08      0.00      0.00     96.50
```

As one can see, for some reason the column `%idle` is either missing entirely or simply moved left by one for some reason and `%gnice` is missing instead. In my opinion the latter is more likely according to the other lines, which always have `0.00` for `%gnice` and an `%idle` of slightly less than `100.00`. Rows like these can not be properly parsed by [ksar and result in error messages](https://github.com/vlsi/ksar/issues/230) like the following:
```
10:33:50.236 [Thread-16] ERROR net.atomique.ksar.graph.Graph - CPU 0 col 11 is missing 00:00:01          0      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
10:33:50.237 [Thread-16] ERROR net.atomique.ksar.graph.Graph - CPU 1 col 11 is missing 00:00:01          1      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
10:33:50.237 [Thread-16] ERROR net.atomique.ksar.graph.Graph - CPU 2 col 11 is missing 00:00:01          2      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
10:33:50.237 [Thread-16] ERROR net.atomique.ksar.graph.Graph - CPU 3 col 11 is missing 00:00:01          3      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
10:33:50.239 [Thread-16] ERROR net.atomique.ksar.graph.Graph - CPU 4 col 11 is missing 00:00:01          4      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
10:33:50.239 [Thread-16] ERROR net.atomique.ksar.graph.Graph - CPU 5 col 11 is missing 00:00:01          5      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
10:33:50.239 [Thread-16] ERROR net.atomique.ksar.graph.Graph - CPU 6 col 11 is missing 00:00:01          6      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
10:33:50.239 [Thread-16] ERROR net.atomique.ksar.graph.Graph - CPU 7 col 11 is missing 00:00:01          7      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
```

**Does anyone know why this happens and how to fix it?** Thanks!

**Comment by sysstat** [comment]
Thanks for reporting this issue.
Please give more details about it: Which version of sysstat/sar are you using? Which command gives this output?
If this problem happens when reading a daily data file (`/var/log/sa/saDD`) then also send me such a data file (sysstat at orange dot fr).

**Comment by ams-tschoening** [comment]
```
$ uname -a
Linux legalxpress 5.4.0-67-generic #75~18.04.1-Ubuntu SMP Tue Feb 23 19:17:50 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
$ sar -V
sysstat version 11.6.1
(C) Sebastien Godard (sysstat <at> orange.fr)
```

The files: [sar.zip](https://github.com/sysstat/sysstat/files/6199034/sar.zip)

```
[...]@[...]:~$ sar -A -f /tmp/sa01 | less
Linux 5.4.0-66-generic (legalxpress)    03/01/2021      _x86_64_        (8 CPU)

12:00:01 AM     CPU      %usr     %nice      %sys   %iowait    %steal      %irq     %soft    %guest    %gnice     %idle
12:00:01 AM     all      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00
12:00:01 AM       0      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
12:00:01 AM       1      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
12:00:01 AM       2      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
12:00:01 AM       3      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
12:00:01 AM       4      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
12:00:01 AM       5      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
12:00:01 AM       6      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
12:00:01 AM       7      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00    100.00
12:05:01 AM     all      4.49      0.00      1.01      6.60      0.00      0.00      0.09      0.00      0.00     87.80
```

Commands were setup by UB 18.04, the only thing I changed was to enable writing files, nothing else.

```
$ cat /etc/default/sysstat
#
# Default settings for /etc/init.d/sysstat, /etc/cron.d/sysstat
# and /etc/cron.daily/sysstat files
#

# Should sadc collect system activity informations? Valid values
# are "true" and "false". Please do not put other values, they
# will be overwritten by debconf!
ENABLED="true"
```

```
$ cat /etc/cron.d/sysstat
# The first element of the path is a directory where the debian-sa1
# script is located
PATH=/usr/lib/sysstat:/usr/sbin:/usr/sbin:/usr/bin:/sbin:/bin

# Activity reports every 10 minutes everyday
*/5 * * * * root command -v debian-sa1 > /dev/null && debian-sa1 1 1

# Additional run at 23:59 to rotate the statistics file
59 23 * * * root command -v debian-sa1 > /dev/null && debian-sa1 60 2
```

```
$ systemctl status sysstat
● sysstat.service - Resets System Activity Data Collector
   Loaded: loaded (/lib/systemd/system/sysstat.service; enabled; vendor preset: enabled)
   Active: active (exited) since Sun 2021-03-21 13:32:20 CET; 3 days ago
     Docs: man:sa1(8)
           man:sadc(8)
           man:sar(1)
  Process: 16035 ExecStart=/usr/lib/sysstat/debian-sa1 --boot (code=exited, status=0/SUCCESS)
 Main PID: 16035 (code=exited, status=0/SUCCESS)
```

**Comment by sysstat** [comment]
Thanks for the details.
This bug has been fixed in sysstat version 12.1.5. See commit c542c25 (the value 0.00 for the %gnice column was actually missing).
So please upgrade.

**Comment by ams-tschoening** [comment]
Thanks for fixing, I asked maintainers of Ubuntu to [backport](https://bugs.launchpad.net/ubuntu/+source/sysstat/+bug/1921747).

**Comment by utkarsh2102** [comment]
Hi @ams-tschoening,

I've prepared a fix for Bionic (Ubuntu 18.04); cf: https://bugs.launchpad.net/ubuntu/+source/sysstat/+bug/1921747 but I've put this on hold due to the issue being non-reproducible. Please see: https://bugs.launchpad.net/ubuntu/+source/sysstat/+bug/1921747/comments/4

Do you think you can reply to that comment and help us in reproducing the issue?
